### DISCUSSION TITLE
Getting up and running

### DISCUSSION THREAD

**Discussion by apar-mistry** [dermatologist/umlsjs#30] [Q&A]
Hello, I am a CS student who's been studying UMLS and came across this package whilst doing my research. Upon installing it with npm, I have not been able to use it even though I have my own API Key. the usage shows what seems to be a constant for an api_key that is most likely a string. I declared a string variable and I would get the error that: 
<img width="429" alt="image" src="https://user-images.githubusercontent.com/73147052/126050171-25458e49-b43e-45ba-b18b-06e7645ca9d7.png">
I would really appreciate it if someone can help me get started with this.

**Comment by dermatologist** [comment]
Hi @apar-mistry 

Thanks for reporting this. I haven't been at this for a while so I am not sure if any of the dependency updates have caused any problems. I have moved this to issues. Will take a look and keep you posted. 

Cheers!

**Reply by apar-mistry** [reply]
Thank you! @dermatologist. Appreciate the quick response.

**Reply by apar-mistry** [reply]
In theory should I have been able to just copy your usage and then create my own const API_KEY and be able to use it?

**Reply by dermatologist** [reply]
@apar-mistry  Yes, absolutely.  

Can you please try adding a **default** in between:
```
import umlsjs from 'umlsjs'
const search1 = new umlsjs.default.UMLSSearch(API_KEY)

```

**Reply by apar-mistry** [reply]
This is what happened after adding that default.
<img width="1125" alt="image" src="https://user-images.githubusercontent.com/73147052/126051251-afdc86e0-3151-4d0b-b39d-1e8241ade068.png">

**Reply by dermatologist** [reply]
Hi @apar-mistry 

Hm, I see. I use localStorage for storing tokens here: https://github.com/dermatologist/umlsjs/blob/develop/src/service/authService.ts
But localStorage is available only in a browser context.

So we need to check if we are in a browser before using localStorage for the cache as shown here:
https://stackoverflow.com/questions/52474208/react-localstorage-is-not-defined-error-showing

On server side we need to use: https://github.com/lmaccherone/node-localstorage
(Or some other strategy for cache)

Do you want to try to fix it and submit a pull request? 🤔

**Reply by apar-mistry** [reply]
Hi @dermatologist , the quick fix I found was to install the package laid out in https://github.com/lmaccherone/node-localstorage. Once you install it, you can use polyfill and get it running. I did that and it works just fine now.

**Reply by dermatologist** [reply]
@apar-mistry Great! 🚀 
If you get time, please submit a PR adding these instructions to README.
I will fix the localStrorage issue and push a release sometime soon!
The *default* in import also needs a fix.
I don't know why this did not show up during testing.

**Reply by apar-mistry** [reply]
Will do!! Thank you so much for the time.

**Reply by dermatologist** [reply]
Hi @apar-mistry
Fixed both in **0.5.0**  (Removed localStorage)

See the new usage instructions in README.

Please test when you have time.

Thanks!

**Reply by apar-mistry** [reply]
Good morning @dermatologist, 

I submitted an issue in regards to the .get functions for CUISearch objects. Please take a look at them when you have the chance. 

Thanks!
